File: javascript/stack.js
var t = -1;
var MAX = 5;
var a = Array(MAX).fill(0); // Maximum size of Stack

function isEmpty() {
  return t < 0;
}

function push(x) {
  if (t >= MAX - 1) {
    console.log("Stack Overflow");
    return false;
  } else {
    t++;
    a[t] = x;

    console.log(x + " pushed into stack");
    return true;
  }
}

function pop() {
  if (t < 0) {
    console.log("Stack UnderFlow");
    return 0;
  } else {
    var x = a[t];
    t--;
    return x;
  }
}

function peek() {
  if (t < 0) {
    console.log("Stack UnderFlow");
    return 0;
  } else {
    var x = a[t];
    return x;
  }
}

function print() {
  for (let i = t; i > -1; i--) {
    console.log("" + a[i]);
  }
}

push(10);
push(20);
push(30);

console.log(pop() + " Popped from stack");
console.log("Top element is :" + peek());
console.log("Elements present in stack : ");
print();

File: python/stack.py
# stack implementation in python

#creating stack




def create_stack():
    stack = []
    return stack


#creating an empty stack


def check_empty(stack):
    return len(stack) == 0


# Adding items into stack


def push(stack, item):
    stack.append(item)
    print("pushed item: " + item)


#removing element from the stack


def pop(stack):
    return "stack is empty" if (check_empty(stack)) else stack.pop()

# peeking element from stack

def peek(stack):
    return "stack is empty" if (check_empty(stack)) else stack[len(stack)-1]


if __name__ == "__main__":
    stack = create_stack()
    push(stack, str(1))
    push(stack, str(2))
    push(stack, str(3))
    push(stack, str(4))
    push(stack, str(5))

    print("popped item: " + pop(stack))
    print("stack after popping an element: " + str(stack))
    print(peek(stack[1]))

